File: android/app/src/main/kotlin/com/cruisemesh/app/notify/ChatMuteStore.kt
package com.cruisemesh.app.notify

import android.content.Context
import android.util.Base64

object ChatMuteStore {
    private const val PREFS = "cruisemesh_chat_mutes"

    fun isMuted(context: Context, chatId: ByteArray): Boolean =
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).getBoolean(key(chatId), false)

    fun setMuted(context: Context, chatId: ByteArray, muted: Boolean) {
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
            .edit().putBoolean(key(chatId), muted).apply()
    }

    private fun key(chatId: ByteArray): String = Base64.encodeToString(chatId, Base64.NO_WRAP or Base64.URL_SAFE)
}

File: ios/CruiseMesh/Notify/ChatMuteStore.swift
import Foundation

enum ChatMuteStore {
    private static let prefix = "cruisemesh.chat.muted."
    static func isMuted(_ chatId: Data) -> Bool { AppDefaults.current.bool(forKey: key(chatId)) }
    static func setMuted(_ muted: Bool, chatId: Data) { AppDefaults.current.set(muted, forKey: key(chatId)) }
    private static func key(_ chatId: Data) -> String { prefix + chatId.base64EncodedString() }
}
